src/cargo/util/toml/mod.rs

```diff
@@ -1737,12 +1738,22 @@ impl TomlManifest {
             debug!("manifest has no build targets");
         }
 
-        if let Err(e) = unique_build_targets(&targets, package_root) {
-            warnings.push(format!(
-                "file found to be present in multiple \
-                 build targets: {}",
-                e
-            ));
+        if let Err(conflict_targets) = unique_build_targets(&targets, package_root) {
+            conflict_targets.iter().for_each(|c| {
+                warnings.push(format!(
+                    "file `{}` found to be present in multiple \
+                 build targets:\n{}",
+                    c.0.display().to_string(),
+                    c.1.iter()
+                        .map(|t| format!(
+                            "  * This file exist in `{}` as a `{}` target.",
+                            t.name(),
+                            t.kind().description()
+                        ))
+                        .collect::<Vec<String>>()
+                        .join("\n")
```

Comment:
Just FYI, itertools can do this in one line (without the call to `collect`), if you import `itertools::Itertools`.

Reply:
Updated. Thanks!